src/main/java/com/rackspace/salus/policy/manage/services/MetadataPolicyManagement.java

```diff
@@ -0,0 +1,231 @@
+            String.format("No policy found with id %s", id)));
+
+    metadataPolicyRepository.deleteById(id);
+    log.info("Removed policy {}", policy);
+    sendMetadataPolicyEvents(policy);
+  }
+
+  /**
+   * Sends metadata policy events for all potentially relevant tenants.
+   * @param policy The MetadataPolicy to distribute out to all tenants.
+   */
+  private void sendMetadataPolicyEvents(MetadataPolicy policy) {
+    log.info("Sending metadata policy events for {}", policy);
+
+    List<String> tenantIds = getTenantsForMetadataPolicy(policy);
+
+    tenantIds.stream()
+        .map(tenantId -> new MetadataPolicyEvent()
+            .setPolicyId(policy.getId())
+            .setTenantId(tenantId))
+        .forEach(policyEventProducer::sendPolicyEvent);
+  }
+
+  /**
+   * Gets all the metadata policies relevant to a tenant.
+   *
+   * Filters the list of all policies down to those that fall into the correct scope/subscope for
+   * the provided tenant.
+   *
+   * @param tenantId The tenantId to retrieve policies for.
+   * @return The list of effective metadata policies that should be applied to the tenant's resources.
+   */
+  public List<MetadataPolicy> getEffectiveMetadataPoliciesForTenant(String tenantId) {
+    return
+        // Create a stream from all metadata policies
+        StreamSupport.stream(metadataPolicyRepository.findAll().spliterator(), false)
+            // Filter the stream for only those policies relevant to this tenant
+            .filter(policy -> policyManagement.isPolicyApplicable(policy, tenantId))
+            // Get one policy for each policy name
+            .collect(
+                // First group the policies by monitor type and key
+                Collectors.groupingBy(policy -> Pair.of(policy.getMonitorType(), policy.getKey()),
+                    // then filter each group by only retrieving the one with the highest priority
+                    Collectors.maxBy(Comparator.comparingInt(policy -> policy.getScope().getPriority())))
+            )
+            // Now we have a map of policy name -> (optional) metadata policy
+            .values().stream()
+            // Filter out any of the optional objects that do not contain anything
+            .filter(Optional::isPresent).map(Optional::get)
+            // Finally convert to a list of the relevant metadata policies
+            .collect(Collectors.toList());
+  }
+
+  /**
+   * Tests whether an equivalent policy already exists.
+   *
+   * @param policy The scope, subscope, and name of this policy will be looked up.
+   * @return True if a policy already exists for these keys, false otherwise.
+   */
+  private boolean exists(MetadataPolicyCreate policy) {
+    return metadataPolicyRepository.existsByScopeAndSubscopeAndMonitorTypeAndKey(
+        policy.getScope(), policy.getSubscope(), policy.getMonitorType(), policy.getKey());
+  }
+
+  /**
+   * Get events for all accounts that the policy may be relevant for.
```

Comment:
Heh, @itzg commented on this line as well and changed `Set` to `Get`.  We both missed the next part.

```suggestion
   * Get a list of tenants that the policy may be relevant for.
```